### PR TITLE
Refactor advection

### PR DESCRIPTION
**Pull request by Peter9192**
This PR

- Rewrites advection as a self-contained function such that it can be more easily tested and changed for another solver
- Adds grid staggering
- Change the handling of periodic boundary by padding cells with ghost cells, such that all local computations in the interior are independent of periodic boundary treatment
- Change accounting of boundary losses: instead of N/E/S/W, just add them to "e_track" in the boundary cells of the domain
- Two small fixes in the analysis code: don't include the first timestep, and use 'robust' plotting so the extremes at the edges scale better.

### CODE REVIEW

**Review by geek-yang — APPROVED** [review]
Nice work! I compare the backtrack results obtained from this PR and those from the main branch. They are exactly the same, which means the changes in the algorithm works.

Only got minor comments related to the docstrings. It would be nice to have equations included in the docstrings, especially for the functions with abstract variable names (e.g. `uq, adv_x`). But they can be added later, perhaps together with unit tests for these functions.

File: src/wam2layers/tracking/backtrack.py [hunk 1]
```diff
@@ -97,68 +98,90 @@ def load_region(config):
...
 
-    # separate directions south-north (all positive numbers)
-    fn_sn = fn * fn_pos
-    fn_ns = fn * fn_neg
 
-    # fluxes over the southern boundary
-    fs_sn = look_south(fn_sn)
-    fs_ns = look_south(fn_ns)
+def pad_boundaries(*args, periodic=False):
+    """Add boundary padding to all input arrays."""
```
[the remark this answers is not in the document]
Comment (Peter9192) [reply src/wam2layers/tracking/backtrack.py]: Nice suggestion :-)

**Comment by Peter9192** [comment]
Added a bit more docstring.

**Comment by Peter9192** [comment]
Ai, it looks like two recent PRs are incompatible (#238 and #265).

**Pull request merged**